todo_app/view_model.py

```diff
@@ -1,7 +1,7 @@
 import datetime
 
 class ViewModel:     
-    def __init__(self, tasks):   
+    def __init__(self, tasks,role='readonly'):   
         self._tasks = tasks   
```

Comment:
You need to store `role` on `self` - at the moment it's just ignored!

Reply:
Done Thanks